src/lib/device/virtual_dev.cpp

```diff
@@ -68,7 +69,7 @@ static std::shared_ptr< BlkAllocator > create_blk_allocator(blk_allocator_type_t
                                   size,
                                   is_auto_recovery,
                                   std::string("varsize_chunk_") + std::to_string(unique_id),
-                                  is_data_drive_hdd() ? false : true /* use_slabs */};
+                                  !is_data_drive_hdd() && use_slab_in_blk_allocator /* use_slabs */};
```

Comment:
This change is fine. 

Now that we have input parameter to set use_slab, ideally, we should move this check `is_data_drive_hdd()` to upper layer (service) to only set use_slab in their parameter? E.g. for meta service, it doesn't care whether there is hdd or not, just set use_slab as false, for other layer like data service, for nublox 2.0 case, data service layer will need to check whether hdd is there or not to set use_slab true or false. 

You can make the decision. If we want to keep this as it is, I think we can have an assert here saying if use_slab is set to true by caller, we should assert is_data_drive_hdd() is false, otherwise, caller would be surprised why I set use_slab to true, but my blk allocator silently set it to false. Make sense?

Reply:
yea, agree, I can add the alert in later pr